packages/grafana-ui/src/components/MatchersUI/FieldNameByRegexMatcherEditor.tsx

```diff
@@ -10,7 +10,7 @@ export const FieldNameByRegexMatcherEditor = memo<MatcherUIProps<string>>(props
     (e: React.FocusEvent<HTMLInputElement>) => {
       return props.onChange(e.target.value);
     },
-    [props.onChange]
+    [props]
```

Comment:
nit: Best practices is to destruct nested props and use them in the dependencies.

Reply:
done

Reply:
not sure I understand, you didn't destruct onChange or you don't agree with me :)